src/App/Fossa/EmbeddedBinary.hs

```diff
@@ -86,13 +88,22 @@ cleanupThemisBins (ThemisBins a b) = traverse_ cleanupExtractedBinaries [unTag a
 withThemisAndIndex :: Has (Lift IO) sig m => (ThemisBins -> m c) -> m c
 withThemisAndIndex = bracket extractThemisFiles cleanupThemisBins
 
+-- When running Themis we always need both the themis-cli and the decompressed index.gob
 extractThemisFiles :: Has (Lift IO) sig m => m ThemisBins
 extractThemisFiles = do
-  themisActual <- applyTag @ThemisBinary <$> extractEmbeddedBinary Themis
-  -- TODO: The gob is still xzipped, unzip it here
-  themisIndex <- applyTag @ThemisIndex <$> extractEmbeddedBinary ThemisIndex
-
-  pure $ ThemisBins themisActual themisIndex
+  themisBin <- extractEmbeddedBinary Themis
+  let themisActual = applyTag @ThemisBinary themisBin
+  -- decompress index.gob.gz binary and write it to disk
+  let bin = toStrict $ Lzma.decompress $ toLazy embeddedBinaryThemisIndex
+  container <- sendIO extractDir
+  let decompressedThemisIndex =
+        BinaryPaths
+          { binaryPathContainer = container
+          , binaryFilePath = $(mkRelFile "index.gob")
+          }
+  _ <- sendIO $ ensureDir $ parent $ toPath decompressedThemisIndex
+  _ <- sendIO $ writeFile (fromAbsFile $ toPath decompressedThemisIndex) bin
```

Comment:
The `from(Abs|Rel)(Dir|File)` functions are only really useful when the `b`/`t` of a `Path b t` aren't known at compile time.  That's not the case here, since `toPath binpath` always returns a `Path Abs File`.

```suggestion
  _ <- sendIO $ writeFile (toString $ toPath decompressedThemisIndex) bin
```